### ISSUE TITLE
[v2] `wails build -platform windows/amd64 -nsis` returns ERROR: cannot build nsis installer - no windows targets

### ISSUE THREAD

**Issue by waltzofpearls** [wailsapp/wails#1714]
### Description

In wails `v2.0.0-beta.42`, when compiling wails app for windows with `wails build -platform windows/amd64 -nsis`, instead of generating a windows installer executable, it outputs an `ERROR: cannot build nsis installer - no windows targets`. In `build/bin` folder, `wails build` only produced `app.exe`, and no `app_installer.exe`.

### To Reproduce

1. In macOS (may be irrelevant to the OS) terminal, go to wails app folder
2. Execute command `wails build -platform windows/amd64 -nsis`
3. In the output towards the end, see `ERROR: cannot build nsis installer - no windows targets`

### Expected behaviour

The output should show:

```
Creating NSIS installer
------------------------------
  - Building 'amd64' installer: Done.
```

And `wails build` should produce both `app.exe` and `app_installer.exe`

### Attempted Fixes

I cloned wails repo locally, and tried to debug wails command. In `v2/cmd/wails/internal/commands/build/build.go` on line 337, when I changed

```go
outputFilename, err := build.Build(buildOptions)
```

to

```go
var err error
outputFilename, err = build.Build(buildOptions)
```

It can successfully generate NSIS installer with `wails build` command.

### System Details

```shell
Wails CLI v2.0.0-beta.42

Scanning system - Please wait (this may take a long time)...Done.

System
------
OS:             MacOS
Version:        12.4
ID:             199506
Go Version:     go1.18.4
Platform:       darwin
Architecture:   arm64

Wails
------
Version:        v2.0.0-beta.42

Dependency                      Package Name    Status          Version
----------                      ------------    ------          -------
xcode command line tools        N/A             Installed       2395
npm                             N/A             Installed       8.1.2
*upx                            N/A             Installed       upx 3.96
*nsis                           N/A             Installed       v3.08

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!


If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony
```

### Additional context

This bug was introduced in PR #1618 and caused by Go's variable shadowing. In that PR, A block of code including line 337 from `v2/cmd/wails/internal/commands/build/build.go` were moved into a `if !dryRun { ... }` block, and changed its behavior to redeclare `outputFilename` as a local variable within the if block.

**Comment by stffabi** [comment]
Thanks for reporting this issue and giving such a detailed analysis 🙏 .

I've pushed PR #1715 which fixes this issue. Would it be possible for you to give it a try?

**Comment by waltzofpearls** [comment]
Hey @stffabi, thanks for the quick reply, and putting together a PR in such a short time.

I tested the fix from PR #1715, and it works!

Thanks again :)
